File: problem003/3.rb
#!/usr/bin/env ruby
n = 600851475143

def factor(n)
    maxfactor = 2
	i = 2
    while i < n/2 do
        if n % i == 0 then
            n = n / i
            maxfactor = n
        else
            i += 1
        end
    end
    return maxfactor
end

puts factor(n)

File: problem003/3.php
#!/usr/bin/env js
<?php
//fail
$n = 600851475143;
//$n = 60;
$i = 2;

function factor($n, $i){
	$maxfactor = 2;
	while ($i < $n/2) {
		if ($n % $i == 0) {
			$n = $n / $i;
			$maxfactor = $n;
		} else {
			$i++;
		}
	}
	return $maxfactor;
}

//print(factor($n, $i));
print("php can't divide big numbers (?)");
?>
